### ISSUE TITLE
Log a single event for exceptions

### ISSUE THREAD

**Issue by seanmsmith23** [ruby-shoryuken/shoryuken#544]
Would it be possible to only log a single event for exceptions?

[Currently](https://github.com/phstc/shoryuken/blob/master/lib/shoryuken/processor.rb#L25-L26) there is one log event being created for the exception message and one log event for the exception backtrace.

This can sometimes make it challenging to debug in a high volume/noisy application as additional tagging is needed to identify the two events as related or they need to be manually identified.

ex: instead of something like: 
```
logger.error { "Processor failed: #{ex.message}" }
logger.error { ex.backtrace.join("\n") } unless ex.backtrace.nil?
```
simplify to:
```
logger.error { "Processor Failed: #{ex.message}. Backtrace: #{ex.backtrace.join("\n")}" }
```

**Comment by phstc** [comment]
@seanmsmith23 that's a good point. I have mixed feelings with changing it. For some people, two lines may work better.

Multiple lines seem to be also [what Sidekiq does](https://github.com/mperham/sidekiq/blob/52562b715174e447a0f7666136838521fda69214/lib/sidekiq/exception_handler.rb#L22-L24).

How are you checking these errors? Are you sending those to an error tracker? What I usually do in my workers, I rescue errors and send them to an error tracking tool. You can also do that using a middleware, so all your workers will behave the same. 

I would love to know more about your use-case. I'm open for changing the code, but we may need to make something customizable, like `processor_error_handler: ExceptionHandler` defaulting to two lines, so we don't introduce "breaking changes" if someone is relying on two lines.

WDYT?

**Comment by seanmsmith23** [comment]
With our setup we changed out the default Shoryuken logger for Ougai. The exceptions are being tracked via New Relic and then typically we'll dig into the logs (via Splunk) for more detail as needed. Currently, it's a bit tough to find the appropriate backtrace given the volume we have. There's a relatively easy intermediate step to take of adding some tagging and uuids to correlate them.

I like the idea of having something to swap out the how exceptions are handled. Or at least to separate out the exception code so that it can be more easily overridden. I briefly considered monkey patching it, but the exception code is coupled to the actual `perform` functionality.

In short: I get not wanting to introduce potentially breaking changes, but it seems nice to have some opportunity to tweak how exceptions are handled.

**Comment by tcostermans** [comment]
This request looks interesting for me as well.
I'm trying to disable the backtrace.

I hope I'm not hijacking this thread, but I also would like a more flexible way to hook into the logging.

I also don't look for breaking changes, 2 lines are ok for me. Based on some config I would like to have the option to disable backtrace logging.

**Comment by phstc** [comment]
@seanmsmith23 @tcostermans 

For avoiding breaking changes, I would rather create an `ExceptionHandler`, letting users customize the log output behavior `Shoryuken.exception_handler = MyExceptionHandler`. [Sidekiq has a flexible implementation for that](https://github.com/mperham/sidekiq/blob/455e9d56f46f0299eaf3b761596207e15f906a39/lib/sidekiq/exception_handler.rb).

```ruby
rescue ex
  # replaces this 
  # logger.error { "Processor failed: #{ex.message}" }
  # logger.error { ex.backtrace.join("\n") } unless ex.backtrace.nil?
  # with 
  Shoryuken.exception_handler.call(ex)
end
```

Would you be interested in submitting a pull request for that? I would be more than happy to review it.

**Comment by nbulaj** [comment]
Done in https://github.com/ruby-shoryuken/shoryuken/pull/742 @phstc
